src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -55,6 +59,32 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
             return [r.name for r in client.load_balancers.get(resource_group_name=rg, load_balancer_name=lb).inbound_nat_rules] # pylint: disable=no-member
     return completer
 
+def get_ag_subresource_completion_list(prop):
+    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
+        client = _network_client_factory()
+        try:
+            ag_name = parsed_args.application_gateway_name
+        except AttributeError:
+            ag_name = parsed_args.resource_name
+        if parsed_args.resource_group_name and ag_name:
+            ag = client.application_gateways.get(parsed_args.resource_group_name, ag_name)
+            return [r.name for r in getattr(ag, prop)]
+    return completer
+
+def get_ag_url_map_rule_completion_list():
+    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
+        client = _network_client_factory()
+        try:
+            ag_name = parsed_args.application_gateway_name
+        except AttributeError:
+            ag_name = parsed_args.resource_name
+        if parsed_args.resource_group_name and ag_name:
+            ag = client.application_gateways.get(parsed_args.resource_group_name, ag_name)
+            url_map = next((x for x in ag.url_path_maps if x.name == parsed_args.url_path_map_name), None) # pylint: disable=no-member
+            if not url_map:
+                raise CLIError('URL path map "{}" not found.'.format(parsed_args.url_path_map_name))
```

Comment:
No, it was copy-paste logic from a validator. It will just flash and do nothing, which is appropriate. 

---
In reply to: [74091861](https://github.com/Azure/azure-cli/pull/616#discussion_r74091861) [](ancestors = 74091861)